compiler/ir/ir.psi2ir/src/org/jetbrains/kotlin/psi2ir/generators/FunctionGenerator.kt

```diff
@@ -289,21 +289,21 @@ class FunctionGenerator(declarationGenerator: DeclarationGenerator) : Declaratio
         }
 
         val bodyGenerator = createBodyGenerator(irFunction.symbol)
+
+        // Declare all the value parameters up first.
         functionDescriptor.valueParameters.mapTo(irFunction.valueParameters) { valueParameterDescriptor ->
             val ktParameter = DescriptorToSourceUtils.getSourceFromDescriptor(valueParameterDescriptor) as? KtParameter
-            generateValueParameterDeclaration(valueParameterDescriptor, ktParameter, bodyGenerator, withDefaultValues, irFunction)
+            declareParameter(valueParameterDescriptor, ktParameter, irFunction)
```

Comment:
@madsager Please add irText test (see  testData/ir/irText/declarations/parameters) for mentioned case

Reply:
Done. I'm unsure if I should add these expectations for the fir2ir test? These expectations say that f2 does not resolve which is incompatible with the current system. Is that OK since fir is so early right now, or should I rather put in some failure marker?

Reply:
Just as a data point: it seems that there are other fir.txt files here that have ERROR_CALL in them where the normal .txt does not.

Reply:
Seems we can keep current variant with IrErrorTypes //cc @mglukhikh @semoro

Reply:
Error types in `.fir.txt` files aren't a problem right now.